code/game/objects/items/devices/body_snatcher_vr.dm

```diff
@@ -0,0 +1,66 @@
+//Body snatcher. Based off the sleevemate, but instead of storing a mind it lets you swap your mind with someone. Extremely illegal and being caught with one s
+/obj/item/device/bodysnatcher
+	name = "\improper Body Snatcher Device"
+	desc = "An extremely illegal tool that allows the user to swap minds with the selected humanoid victim. "
+	icon = 'icons/obj/device_alt.dmi'
+	icon_state = "sleevemate" //Give this a fancier sprite later.
+	item_state = "healthanalyzer"
+	flags = CONDUCT
+	slot_flags = SLOT_BELT
+	w_class = ITEMSIZE_SMALL
+	matter = list(DEFAULT_WALL_MATERIAL = 200)
+	origin_tech = list(TECH_MAGNET = 2, TECH_BIO = 2, TECH_ILLEGAL = 1)
+
+/obj/item/device/bodysnatcher/New()
+	..()
+	flags |= NOBLUDGEON //So borgs don't spark.
+
+/obj/item/device/bodysnatcher/attack(mob/living/M, mob/living/user)
+	usr.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+	if(ishuman(M) || issilicon(M)) //Allows body swapping with humans, synths, and pAI's/borgs since they all have a mind.
+		if(!M.mind) //Do they have a mind?
+			to_chat(usr,"<span class='warning'>A warning pops up on the device, informing you that [M] appears braindead.</span>")
+			return
+		if(M.stat == DEAD) //Are they dead?
+			to_chat(usr,"<span class='warning'>A warning pops up on the device, informing you that [M] is dead, and, as such, the mind transfer can not be done.</span>")
+			return
+		var/choice = alert(usr,"This will swap your mind with the target's mind. This will result in them controlling your body, and you controlling their body. Continue?","Confirmation","Continue","Cancel")
+		if(choice == "Continue" && usr.get_active_hand() == src && usr.Adjacent(M))
+
+			usr.visible_message("<span class='warning'>[usr] pushes the device up his forehead and [M]'s head, the device beginning to let out a series of light beeps!</span>","<span class='notice'>You begin swap minds with [M]!</span>")
+			if(do_after(usr,35 SECONDS,M))
+				if(usr.mind && M.mind && M.stat != DEAD && usr.stat != DEAD)
+					//Most of this could be done with a ghostize and simple M.ckey = usr.ckey, but then it has a chance of giving admins the wrong CID and IP, thus, the code below.
+					var/taget_cid = M.computer_id
+					var/taget_ip = M.lastKnownIP
+					var/target_ckey = M.ckey
+					var/target_ooc_notes = M.ooc_notes
+					var/user_cid = usr.computer_id
+					var/user_ip = usr.lastKnownIP
+					var/user_ckey = usr.ckey
+					var/user_ooc_notes = user.ooc_notes
+					admin_attack_log("[usr.ckey] used a Bodysnatcher to swap bodies with [M.ckey]")
+					to_chat(usr,"<span class='notice'>Your minds have been swapped! Have a nice day.</span>")
+					M.ghostize() //If this doesn't happen, it boots them from the server.
+					M.ckey = user_ckey
+					usr.ckey = target_ckey
+					M.computer_id = user_cid
+					usr.computer_id = taget_cid
+					M.lastKnownIP = user_ip
```

Comment:
Ah, alright. Currently looking at resleeving/machines.dm, and I think I see how I can apply the mind transfer to this. Making DNM until I change it.